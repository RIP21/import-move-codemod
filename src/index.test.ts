import pluginTester from "babel-plugin-tester";
import transform, { Config } from "./index";

const { format } = require("prettier");

pluginTester({
  pluginName: "import-rename-codemod",
  plugin: transform,
  snapshot: true,
  formatResult: (output: any) =>
    format(output, {
      semi: true,
      singleQuote: true,
      parser: "babel",
    }),
  tests: [
    {
      title:
        "When named to named import and only one imported should change current import",
      code: `
      import { One } from 'one'
      `,
      pluginOptions: {
        module: {
          from: "one",
          to: "two",
        },
        specifiers: ["One"],
      } as Config,
    },
    {
      title:
        "When named to named import and multiple imported and only one identifier moved it should keep one import and create new import",
      code: `
      import { One, Two } from 'one'
      `,
      pluginOptions: {
        module: {
          from: "one",
          to: "two",
        },
        specifiers: ["Two"],
      } as Config,
    },
    {
      title:
        "When named to named import and multiple imported and all moved it should keep and change import",
      code: `
      import { One, Two } from 'one'
      `,
      pluginOptions: {
        module: {
          from: "one",
          to: "two",
        },
        specifiers: ["One", "Two"],
      } as Config,
    },
    {
      title:
        "When named to named with 'as' import and multiple imported and all moved it should keep and change import",
      code: `
      import { One as OneA, Two as TwoA } from 'one'
      `,
      pluginOptions: {
        module: {
          from: "one",
          to: "two",
        },
        specifiers: ["One", "Two"],
      } as Config,
    },
    {
      title: "When named to named and imports should be renamed",
      code: `
      import { One, Two } from 'one'
      `,
      pluginOptions: {
        module: {
          from: "one",
          to: "two",
        },
        specifiers: {
          One: "OneR",
          Two: "TwoR",
        },
      } as Config,
    },
    {
      title:
        "When named to named and imports should be renamed and default should be preserved",
      code: `
      import ADefault, { One, Two } from 'one'
      `,
      pluginOptions: {
        module: {
          from: "one",
          to: "two",
        },
        specifiers: {
          One: "OneR",
          Two: "TwoR",
        },
      } as Config,
    },
    {
      title: "When named to default should change import",
      code: `
      import { One } from 'one'
      `,
      pluginOptions: {
        module: {
          from: "one",
          to: "two",
        },
        specifiers: {
          One: "default",
        },
      } as Config,
    },
    {
      title: "When wildcard should move all imports",
      code: `
      import ADefault, { One, Two, Three } from 'one'
      `,
      pluginOptions: {
        module: {
          from: "one",
          to: "two",
        },
        specifiers: ["*"],
      } as Config,
    },
    {
      title: "Should move only default import",
      code: `
      import ADefault, { One, Two, Three } from 'one'
      `,
      pluginOptions: {
        module: {
          from: "one",
          to: "two",
        },
        specifiers: ["default"],
      } as Config,
    },
    {
      title: "Should move default import and named imports",
      code: `
      import ADefault, { One, Two, Three } from 'one'
      `,
      pluginOptions: {
        module: {
          from: "one",
          to: "two",
        },
        specifiers: ["default", "One", "Two"],
      } as Config,
    },
    {
      title: "Should move default to named",
      code: `
      import One from 'one'
      `,
      pluginOptions: {
        module: {
          from: "one",
          to: "two",
        },
        specifiers: { default: "One" },
      } as Config,
    },
    {
      title: "Should move default to named while keeping others that are left",
      code: `
      import One, { Two } from 'one'
      `,
      pluginOptions: {
        module: {
          from: "one",
          to: "two",
        },
        specifiers: { default: "One" },
      } as Config,
    },
    {
      title: "Should move default to named and move others along",
      code: `
      import One, { Two } from 'one'
      `,
      pluginOptions: {
        module: {
          from: "one",
          to: "two",
        },
        specifiers: { default: "One", Two: "Two" },
      } as Config,
    },
    {
      title: "Should move default to default",
      code: `
      import One from 'one'
      `,
      pluginOptions: {
        module: {
          from: "one",
          to: "two",
        },
        specifiers: ["default"],
      } as Config,
    },
    {
      title:
        "Should fail when asked to create impossible two or more defaults from named imports",
      code: `
      import One from 'one'
      `,
      pluginOptions: {
        module: {
          from: "one",
          to: "two",
        },
        specifiers: { One: "default", Two: "default" },
      } as Config,
      error:
        "It's impossible to make two named imports into two defaults! You should only have one 'default' value in your specifiers mapping",
    },
    {
      title: "Should fail when provided with wrong config shape",
      code: `
      import One from 'one'
      `,
      pluginOptions: {
        module: {
          from: "one",
          to: "two",
        },
        specifiers: "Wrong",
      },
      error:
        "Wrong specifiers format provided! It should be non-empty object or array.",
    },
  ],
});
