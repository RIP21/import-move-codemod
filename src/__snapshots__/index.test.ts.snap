// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import-rename-codemod Should move default import and named imports: Should move default import and named imports 1`] = `

import ADefault, { One, Two, Three } from 'one'

      ↓ ↓ ↓ ↓ ↓ ↓

import ADefault, { One, Two } from 'two';
import { Three } from 'one';


`;

exports[`import-rename-codemod Should move default to default: Should move default to default 1`] = `

import One from 'one'

      ↓ ↓ ↓ ↓ ↓ ↓

import One from 'two';


`;

exports[`import-rename-codemod Should move default to named and move others along: Should move default to named and move others along 1`] = `

import One, { Two } from 'one'

      ↓ ↓ ↓ ↓ ↓ ↓

import { One, Two } from 'two';


`;

exports[`import-rename-codemod Should move default to named while keeping others that are left: Should move default to named while keeping others that are left 1`] = `

import One, { Two } from 'one'

      ↓ ↓ ↓ ↓ ↓ ↓

import { One } from 'two';
import { Two } from 'one';


`;

exports[`import-rename-codemod Should move default to named: Should move default to named 1`] = `

import One from 'one'

      ↓ ↓ ↓ ↓ ↓ ↓

import { One } from 'two';


`;

exports[`import-rename-codemod Should move only default import: Should move only default import 1`] = `

import ADefault, { One, Two, Three } from 'one'

      ↓ ↓ ↓ ↓ ↓ ↓

import ADefault from 'two';
import { One, Two, Three } from 'one';


`;

exports[`import-rename-codemod Should not create useless default import if 'from' library import is present but has no required named import: Should not create useless default import if 'from' library import is present but has no required named import 1`] = `

import SomeImport from 'one'

      ↓ ↓ ↓ ↓ ↓ ↓

import SomeImport from 'one';


`;

exports[`import-rename-codemod Should not produce useless import if no match when default and named specifiers: Should not produce useless import if no match when default and named specifiers 1`] = `

import { Three } from 'one'

      ↓ ↓ ↓ ↓ ↓ ↓

import { Three } from 'one';


`;

exports[`import-rename-codemod Should not produce useless import if no match: Should not produce useless import if no match 1`] = `

import One, { Two } from 'one'

      ↓ ↓ ↓ ↓ ↓ ↓

import One, { Two } from 'one';


`;

exports[`import-rename-codemod When named to default should change import: When named to default should change import 1`] = `

import { One } from 'one'

      ↓ ↓ ↓ ↓ ↓ ↓

import One from 'two';


`;

exports[`import-rename-codemod When named to named and imports should be renamed and default should be preserved: When named to named and imports should be renamed and default should be preserved 1`] = `

import ADefault, { One, Two } from 'one'

      ↓ ↓ ↓ ↓ ↓ ↓

import { OneR, TwoR } from 'two';
import ADefault from 'one';


`;

exports[`import-rename-codemod When named to named and imports should be renamed: When named to named and imports should be renamed 1`] = `

import { One, Two } from 'one'

      ↓ ↓ ↓ ↓ ↓ ↓

import { OneR, TwoR } from 'two';


`;

exports[`import-rename-codemod When named to named import and multiple imported and all moved it should keep and change import: When named to named import and multiple imported and all moved it should keep and change import 1`] = `

import { One, Two } from 'one'

      ↓ ↓ ↓ ↓ ↓ ↓

import { One, Two } from 'two';


`;

exports[`import-rename-codemod When named to named import and multiple imported and only one identifier moved it should keep one import and create new import: When named to named import and multiple imported and only one identifier moved it should keep one import and create new import 1`] = `

import { One, Two } from 'one'

      ↓ ↓ ↓ ↓ ↓ ↓

import { Two } from 'two';
import { One } from 'one';


`;

exports[`import-rename-codemod When named to named import and only one imported should change current import: When named to named import and only one imported should change current import 1`] = `

import { One } from 'one'

      ↓ ↓ ↓ ↓ ↓ ↓

import { One } from 'two';


`;

exports[`import-rename-codemod When named to named with 'as' import and multiple imported and all moved it should keep and change import: When named to named with 'as' import and multiple imported and all moved it should keep and change import 1`] = `

import { One as OneA, Two as TwoA } from 'one'

      ↓ ↓ ↓ ↓ ↓ ↓

import { One as OneA, Two as TwoA } from 'two';


`;

exports[`import-rename-codemod When wildcard should move all imports: When wildcard should move all imports 1`] = `

import ADefault, { One, Two, Three } from 'one'

      ↓ ↓ ↓ ↓ ↓ ↓

import ADefault, { One, Two, Three } from 'two';


`;
